src/AggregateFunctions/UniqExactSet.h

```diff
@@ -27,6 +27,56 @@ class UniqExactSet
             asTwoLevel().insert(std::forward<Arg>(arg));
     }
 
+    /// In merge, if one of the lhs and rhs is twolevelset and the other is singlelevelset, then the singlelevelset will need to convertToTwoLevel().
+    /// It's not in parallel and will cost extra large time if the thread_num is large.
+    /// This method will convert all the SingleLevelSet to TwoLevelSet in parallel if the hashsets are not all singlelevel or not all twolevel.
+    static void parallelizeMergePrepare(const std::vector<UniqExactSet *> & data_vec, ThreadPool * thread_pool = nullptr)
```

Comment:
could we make sure that `ThreadPool` is always not null here and pass it as a reference?

Reply:
From the `void parallelizeMergePrepare(AggregateDataPtrs & places, ThreadPool & thread_pool) const override` in `AggregateFunctionUniq.h`, the `ThreadPool` is alway not null. The latest code has used the reference instead of a pointer.